frontend/lendme/lib/screens/other/add_item.dart

```diff
@@ -1,26 +1,206 @@
+import 'dart:io';
+import 'package:firebase_storage/firebase_storage.dart';
 import 'package:flutter/material.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:permission_handler/permission_handler.dart';
 
-class AddItem extends StatelessWidget {
+class AddItem extends StatefulWidget {
   const AddItem({Key? key}) : super(key: key);
 
+  @override
+  _AddItemState createState() => _AddItemState();
+}
+
+class _AddItemState extends State<AddItem> {
+  final FirebaseAuth firebaseAuth = FirebaseAuth.instance;
+  CollectionReference items = FirebaseFirestore.instance.collection('items');
+  String description = '';
+  String itemName = '';
+  var imageUrl = null;
+  var localImage;
+  bool isLoading = false;
+
   @override
   Widget build(BuildContext context) {
+    Future getImage() async {
+      final _storage = FirebaseStorage.instance;
+      File _image;
+      final _picker = ImagePicker();
+      //Check Permissions
+      await Permission.photos.request();
+      var permissionStatus = await Permission.photos.status;
+      if (permissionStatus.isGranted) {
+        //Select Image
+        final image = await _picker.pickImage(source: ImageSource.gallery);
+
+        setState(() {
+          isLoading = true;
+        });
+        if (image != null) {
+          _image = File(image.path);
+        } else {
+          isLoading = false;
+          return;
+        }
+        //Upload to Firebase
+        var snapshot =
+            await _storage.ref().child('images/items/itemName').putFile(_image);
```

Comment:
To solve problem with storing all images unser same path you could write document to firestore without imageUrl, then get id assigned to this document, upload image to storage with this id as name, and then update imageUrl field inside document, but to keep is simple I suggest [this approach](https://stackoverflow.com/a/61069183/7762960)